hal/src/common/thumbv6m/sercom/v1/uart.rs

```diff
@@ -286,6 +286,34 @@ macro_rules! uart {
                 unsafe fn usart(&self) -> &USART {
                     return &self.sercom.usart();
                 }
+
+                pub fn intenset<F>(&mut self, f: F)
+                where F: FnOnce(&mut crate::target_device::sercom0::usart::intenset::W)
+                {
+                    unsafe {
+                        self.usart().intenset.write(|w| {
+                            f(w);
+                            w
+                        });
+                    }
+                }
+
+                pub fn intenclr<F>(&mut self, f: F)
+                where F: FnOnce(&mut crate::target_device::sercom0::usart::intenclr::W)
+                {
+                    unsafe {
+                        self.usart().intenclr.write(|w| {
+                            f(w);
+                            w
+                        });
+                    }
+                }
+
+                pub fn flags(&self) -> crate::target_device::sercom0::usart::status::R {
+                    unsafe {
+                        self.usart().status.read()
+                    }
+                }
```

Comment:
I'm not a huge fan of this approach. I'd rather keep the direct calls to PAC code hidden and use something like `biitflags` or some other crate to implement the flags. That's the approach I took in `v2::spi`.

But at the same time, this is a `v1` module, and someone may come along and create a more flexible `v2` module. We can probably be more discerning about a new API and more liberal with the changes we make to the existing API.

Reply:
This is the exact API that's already exposed by the `thumbv7em` version of the package. I don't disagree that it's non-ideal but I think consistency is more important.